Sources/MapLibreSwiftUI/Extensions/MapView/MapViewGestures.swift

```diff
@@ -15,6 +15,17 @@ extension MapView {
             let gestureRecognizer = UITapGestureRecognizer(target: context.coordinator,
                                                            action: #selector(context.coordinator.captureGesture(_:)))
             gestureRecognizer.numberOfTapsRequired = numberOfTaps
+            if numberOfTaps == 1 {
+                // If a user double taps to zoom via the built in gesture, a normal
+                // tap should not be triggered.
+                if let doubleTapRecognizer = mapView.gestureRecognizers?
+                    .first(where: {
+                        $0 is UITapGestureRecognizer && ($0 as! UITapGestureRecognizer).numberOfTapsRequired == 2
+                    })
+                {
+                    gestureRecognizer.require(toFail: doubleTapRecognizer)
+                }
+            }
```

Comment:
Yeah, this is a very specific fix and I didn't go deeper to check if there's any other setups we would need to check for.

MapLibre comes with 8 built in gestures, which also have own "require to fail" dependencies:

```
(lldb) po mapView.gestureRecognizers
▿ Optional<Array<UIGestureRecognizer>>
  ▿ some : 8 elements
    - 0 : <UIPanGestureRecognizer: 0x1056231f0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handlePanGesture:, target=<MLNMapView 0x1078d6800>)>; must-fail-for = {
        <UIPanGestureRecognizer: 0x105623910; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerDragGesture:, target=<MLNMapView 0x1078d6800>)>>
    }>
    - 1 : <UIPinchGestureRecognizer: 0x1056233a0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handlePinchGesture:, target=<MLNMapView 0x1078d6800>)>; must-fail-for = {
        <UITapGestureRecognizer: 0x105623cc0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTouchesRequired = 2>
    }>
    - 2 : <UIRotationGestureRecognizer: 0x105623540; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleRotateGesture:, target=<MLNMapView 0x1078d6800>)>; must-fail-for = {
        <UITapGestureRecognizer: 0x105623cc0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTouchesRequired = 2>
    }>
    - 3 : <UITapGestureRecognizer: 0x1056236a0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleDoubleTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTapsRequired = 2; must-fail-for = {
        <UITapGestureRecognizer: 0x1056243c0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleSingleTapGesture:, target=<MLNMapView 0x1078d6800>)>>,
        <UILongPressGestureRecognizer: 0x105623de0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleQuickZoomGesture:, target=<MLNMapView 0x1078d6800>)>>
    }>
    - 4 : <UIPanGestureRecognizer: 0x105623910; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerDragGesture:, target=<MLNMapView 0x1078d6800>)>; must-fail = {
        <UIPanGestureRecognizer: 0x1056231f0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handlePanGesture:, target=<MLNMapView 0x1078d6800>)>>
    }; must-fail-for = {
        <UITapGestureRecognizer: 0x105623cc0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTouchesRequired = 2>
    }>
    - 5 : <UITapGestureRecognizer: 0x105623cc0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTouchesRequired = 2; must-fail = {
        <UIPinchGestureRecognizer: 0x1056233a0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handlePinchGesture:, target=<MLNMapView 0x1078d6800>)>>,
        <UIPanGestureRecognizer: 0x105623910; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleTwoFingerDragGesture:, target=<MLNMapView 0x1078d6800>)>>,
        <UIRotationGestureRecognizer: 0x105623540; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleRotateGesture:, target=<MLNMapView 0x1078d6800>)>>
    }>
    - 6 : <UILongPressGestureRecognizer: 0x105623de0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleQuickZoomGesture:, target=<MLNMapView 0x1078d6800>)>; must-fail = {
        <UITapGestureRecognizer: 0x1056236a0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleDoubleTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTapsRequired = 2>
    }; must-fail-for = {
        <UITapGestureRecognizer: 0x1056243c0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleSingleTapGesture:, target=<MLNMapView 0x1078d6800>)>>
    }>
    - 7 : <UITapGestureRecognizer: 0x1056243c0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleSingleTapGesture:, target=<MLNMapView 0x1078d6800>)>; must-fail = {
        <UITapGestureRecognizer: 0x1056236a0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleDoubleTapGesture:, target=<MLNMapView 0x1078d6800>)>; numberOfTapsRequired = 2>,
        <UILongPressGestureRecognizer: 0x105623de0; state = Possible; view = <MLNMapView: 0x1078d6800>; target= <(action=handleQuickZoomGesture:, target=<MLNMapView 0x1078d6800>)>>
    }>
```

Since we don't want to expose the gesture recognizers or MapView to the user, I feel we have to help set up the gestures that a user adds to work with the built in gestures - just like SwiftUI where the motto is "convention over configuration" - the defaults should cover the most likely usage scenarios, ideally with a way of opting out of default behavior. So maybe we should add a further flag called "requireBuiltInGesturesToFail" which per default is true, but a user can set to false if they'd like their gestures to trigger regardless of the built in ones.

There are actually a total of 3 built in gesture recognizers that have `numberOfTapsRequired = 2`, so more code might be required, but in my testing, this code fixed the "user double tapped the map to zoom in, and yet the onMapGesture closure was run" problem, so I opted for the least invasive implementation for now.